src/components/AnimatedModal/index.tsx

```diff
@@ -14,30 +29,52 @@ import { ANIMATION_DURATION, MAX_VISIBILITY } from '../../core/constants/data';
 const AnimatedPressable = Animated.createAnimatedComponent( Pressable );
 const HEIGHT = Dimensions.get( 'window' ).height;
 
-const AnimatedModal = forwardRef<AnimatedModalPublicMethods, AnimatedModalProps>( ( {
-  children,
-  visible = false,
-  onShow,
-  onHide,
-  closeOnEmptySpace = true,
-  closeOnPressBack = true,
-  animationDuration = ANIMATION_DURATION,
-  closeSpaceVisibility = MAX_VISIBILITY,
-  hideKeyboardOnShow = true,
-  useKeyboardAvoidingView = true,
-}, ref ) => {
-
-  const [ isVisible, setIsVisible ] = useState( visible );
-
-  const animation = useSharedValue( visible ? 1 : 0 );
-  const opened = useSharedValue( visible );
-
-  const animatedPressableStyle = useAnimatedStyle( () => (
-    { opacity: interpolate( animation.value, [ 0, 1 ], [ 0, closeSpaceVisibility ] ) }
-  ), [] );
-  const animatedStyle = useAnimatedStyle( () => (
-    { top: interpolate( animation.value, [ 0, 1 ], [ HEIGHT, 0 ] ) }
-  ), [] );
+// eslint-disable-next-line max-len
+const AnimatedModal = forwardRef<AnimatedModalPublicMethods, AnimatedModalProps>( ( props, ref ) => {
+
+  const {
+    children,
+    onShow,
+    onHide,
+    closeOnEmptySpace = true,
+    closeOnPressBack = true,
+    animationDuration = ANIMATION_DURATION,
+    closeSpaceVisibility = MAX_VISIBILITY,
+    hideKeyboardOnShow = true,
+    useKeyboardAvoidingView = true,
+  } = props;
+
+  // default visibility should never change, hence useMemo with empty dependency array
+  // eslint-disable-next-line max-len
+  const defaultVisibility: boolean = useMemo( () => ( props.controlled ? props.open : ( props.visible ?? false ) ), [] );
```

Comment:
i would remove props.controlled and just check if props.open exists, so user no need to define controlled prop.
`const defaultVisibility: boolean = useMemo( () => ( typeof props.open !== 'undefined' ? props.open : ( props.visible ?? false ) ), [] );`

Reply:
will do now